server/service/src/invoice_line/inbound_shipment_line/update/generate.rs

```diff
@@ -85,23 +85,21 @@ fn generate_line(
     update_line.number_of_packs = number_of_packs
         .map(u32_to_i32)
         .unwrap_or(update_line.number_of_packs);
+    update_line.tax = tax.map(|tax| tax.percentage).unwrap_or(update_line.tax);
 
     if let Some(item) = new_item_option {
         update_line.item_id = item.id;
         update_line.item_code = item.code;
         update_line.item_name = item.name;
     }
 
-    if let Some(total_before_tax) = total_before_tax {
-        update_line.total_before_tax = total_before_tax;
+    update_line.total_before_tax = if let Some(total_before_tax) = total_before_tax {
+        total_before_tax
+    } else if number_of_packs.is_some() || cost_price_per_pack.is_some() {
+        update_line.cost_price_per_pack * update_line.number_of_packs as f64
     } else {
-        update_line.total_before_tax =
-            update_line.cost_price_per_pack * update_line.number_of_packs as f64;
-    }
-
-    if let Some(tax) = tax {
-        update_line.tax = tax.percentage;
-    }
+        update_line.cost_price_per_pack * update_line.number_of_packs as f64
+    };
```

Comment:
why the `if` here? just use the else?

nit: use same cost_price_per_pack/number_of_packs order as in the if statement